packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
+      if (identity == null) {
+        _logger.printError(
+          'Saved signing certificate "$savedCertChoice" is not a valid development certificate. To clear, use "flutter config"',
+        );
+      }
+    }
+
+    if (identity == null) {
+      identity = await _selectSigningIdentity(
+        validCodeSigningIdentities,
+        autoSelectSingle: true,
+        throwOnCancel: true,
+      );
+      if (identity != null) {
+        _saveCodeSignIdentity(identity);
+      } else {
+        return null;
+      }
+    }
+
+    final String? developmentTeam = await _getDevelopmentTeamFromIdentity(identity);
+    if (developmentTeam == null) {
+      return null;
+    }
+    _logger.printStatus('Developer identity "$identity" selected for iOS code signing');
+    return <String, String>{_developmentTeamBuildSettingName: developmentTeam};
   }
 
-  const List<String> findIdentityCommand = <String>[
-    'security',
-    'find-identity',
-    '-p',
-    'codesigning',
-    '-v',
-  ];
-
-  String findIdentityStdout;
-  try {
-    findIdentityStdout =
-        (await processUtils.run(findIdentityCommand, throwOnError: true)).stdout.trim();
-  } on ProcessException catch (error) {
-    logger.printTrace('Unexpected failure from find-identity: $error.');
-    return null;
+  void _saveCodeSignIdentity(String identity) {
+    _logger.printStatus('Certificate choice "$identity" saved.');
+    _config.setValue(kConfigCodeSignCertificate, identity);
   }
 
-  final List<String> validCodeSigningIdentities =
-      findIdentityStdout
-          .split('\n')
-          .map<String?>((String outputLine) {
-            return _securityFindIdentityDeveloperIdentityExtractionPattern
-                .firstMatch(outputLine)
-                ?.group(1);
-          })
-          .where(_isNotEmpty)
-          .whereType<String>()
-          .toSet() // Unique.
-          .toList();
-
-  final String? signingIdentity = await _chooseSigningIdentity(
-    validCodeSigningIdentities,
-    logger,
-    config,
-    terminal,
-    shouldExitOnNoCerts,
-  );
+  void _saveProvisioningProfile(_ProvisioningProfile profile) {
+    _logger.printStatus('Provisioning Profile "${profile.name}" saved.');
+    _config.setValue(kConfigCodeSignProvisioningProfile, profile.filePath);
+  }
 
-  // If none are chosen, return null.
-  if (signingIdentity == null) {
-    return null;
+  /// Validates that command-line tools `security` and `openssl` are available.
+  Future<bool> _validateCodeSignSearchTools({bool printError = false}) async {
+    // If the user's environment is missing the tools needed to find and read
+    // certificates, abandon. Tools should be pre-equipped on macOS.
+    if (!await _processUtils.exitsHappy(const <String>['which', 'security']) ||
+        !await _processUtils.exitsHappy(const <String>['which', 'openssl'])) {
+      if (printError) {
+        _logger.printError('Unable to validate code-signing tools.');
```

Comment:
Consider clarifying that `security` and `openssl` are required in this error message.